dao/src/main/scala/za/co/absa/enceladus/dao/MenasRestDAO.scala

```diff
@@ -136,17 +132,21 @@ object MenasRestDAO extends MenasDAO {
         val status = response.getStatusLine.getStatusCode
         val ok = status >= HttpStatus.SC_OK && status < HttpStatus.SC_MULTIPLE_CHOICES
         val unAuthorized = status == HttpStatus.SC_UNAUTHORIZED
-        if (ok) {
-          log.info(response.toString)
-        }
-        else if (unAuthorized) {
-          throw new UnauthorizedException
-        }
-        else {
-          val responseBody = getResponseBody(response)
-          log.error(s"RESPONSE: ${response.getStatusLine} - $responseBody")
+        if (unAuthorized) {
+          log.warn(s"Unauthorized POST request for Menas URL: $url")
+          log.warn(s"Expired session, reauthenticating")
+          enceladusLogin()
+          log.info(s"Retrying POST request for Menas URL: $url")
+          sendPostJson(url, json)
+        } else {
```

Comment:
Is there a need for this else?

Reply:
Yeah, otherwise we get into the other case as well after a successful retry, which logs an error for the original failed try at `146` and returns `ok` at `148`

Reply:
I meant to make it else if